src/machine-learning/machine-learning-api-client.ts

```diff
@@ -37,17 +37,28 @@ export interface StatusErrorResponse {
 }
 
 /**
- * A Firebase ML Model input object
+ * Firebase ML Model input objects
  */
-export interface ModelOptions {
+export interface ModelOptionsBase {
```

Comment:
Same here. Just call this ModelOptions and extend it where necessary. Get rid of the union types.

Reply:
discussed offline.